platform/commonUI/edit/src/actions/SaveInProgressDialog.js

```diff
@@ -1,21 +1,30 @@
 define([], function () {
-    function SaveInProgressDialog(dialogService) {
-        this.dialogService = dialogService;
-        this.dialog = undefined;
+    var self = this;
```

Comment:
I believe `this` will be the `window` object here, and it seems ill-advised to use this as the this context for the object, especially as it will be shared between instances. It's also inconsistent with the rest of our code. There are a couple of different approaches we use throughout the code for addressing the loss of the `this` context with callbacks, which I'll mention inline below.